packages/frontend/src/components/ExecutionRow/index.tsx

```diff
@@ -63,7 +63,11 @@ export default function ExecutionRow(props: ExecutionRowProps): ReactElement {
           >
             <GridItem area="apps">
               <HStack>
-                <FlowAppIcons steps={flow.steps} />
+                <FlowAppIcons
+                  steps={[...executionSteps].sort(
```

Comment:
can add a comment here that we are passing executionSteps into steps prop to preserve deleted execution steps